backend/src/test/java/com/google/backend/servlets/LoginServletTest.java

```diff
@@ -0,0 +1,131 @@
+package com.google.backend.servlets;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito.*;
+import org.mockito.MockitoAnnotations;
+
+/** Contain tests for methods in {@link LoginServlet} class. */
+@RunWith(JUnit4.class)
+public class LoginServletTest {
+  @Mock HttpServletRequest request;
+  @Mock HttpServletResponse response;
+
+  private static final LoginServlet loginServlet = new LoginServlet();
+  StringWriter stringWriter = new StringWriter();
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper(
+    new LocalDatastoreServiceTestConfig(), new LocalUserServiceTestConfig());
+
+  private static final String AUTH_DOMAIN_NAME = "localhost";
+  private static final String CONTENT_TYPE = "text/html;";
+  private static final String PROPERTY_NAME = "email";
+  private static final String TEST_EMAIL = "test@example.com";
+  private static final String REDIRECT_URL = "/";
+  private static final String LOGIN_STATUS = "logged in";
+  private static final String LOGOUT_STATUS = "stranger";
+  private static final String ENTITY_KIND = "User";
+  private static final int MAX_QUERIES = 1000;
```

Comment:
put all the private static final together in the top, then private final, then everything else.
Also, StringWriter can be private static final as well?

Reply:
StringWriter can't be private static final because each test needs a new StringWriter, I made it private though. 
if it is static final then the previous responses are still in the buffer i believe. I could be wrong about this though